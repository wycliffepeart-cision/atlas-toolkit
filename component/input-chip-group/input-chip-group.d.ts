export declare class InputChipGroup extends HTMLElement {
}
//# sourceMappingURL=input-chip-group.d.ts.map