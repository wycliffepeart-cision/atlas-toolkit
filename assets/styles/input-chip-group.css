@layer atlas {
	atlas-input-chip-group {
		--inset-size: 1em;

		--wrapper-border-radius: var(--atlas-input-common-border-radius);
		--wrapper-border: var(--atlas-input-common-border);

		--chips-label-font-size: var(--atlas-input-common-label-font-size);
		--chips-label-color: var(--atlas-input-common-label-color);
		--chips-label-padding-before: 0.5em;
		--chips-label-padding-after: 0.55em;
		--chips-label-margin-top: 1.2em;

		--chips-min-height: calc(var(--atlas-chip-body-height) * 5);
		--chips-max-height: calc(var(--atlas-chip-body-height) * 9);

		--suggestion-min-height: calc(var(--atlas-chip-body-height) * 2);
		--suggestion-max-height: calc(var(--atlas-chip-body-height) * 5);
		--suggestion-box-shadow: var(--atlas-input-common-box-shadow);
		--suggestion-inline-padding: 0.5em;

		--overall-control-padding: 0.5em;
		--overall-margin-block-end: 3em;

		--backdrop-opacity: 0.7;
		--backdrop-height: 100%;

		display: grid;
		grid-template-columns: 1fr;
		grid-template-areas:
			"label"
			"input"
			"suggestion"
			"controls"
			"chips-label"
			"chips";

		width: 100%;
		padding-block: var(--overall-control-padding);
		margin-block-end: var(--overall-margin-block-end);
	}

	atlas-input-chip-group::part(label) {
		grid-area: label;
		font-size: var(--atlas-input-common-label-font-size);
	}

	atlas-input-chip-group > [slot="label"] {
		display: block;
		margin-block-end: var(--atlas-input-common-label-block-spacing);
	}

	atlas-input-chip-group::part(chips-label) {
		grid-area: chips-label;
		display: block;
		color: var(--chips-label-color);
		border-inline: var(--wrapper-border);
		border-block-start: var(--wrapper-border);
		padding-block-start: var(--chips-label-padding-before);
		padding-block-end: var(--chips-label-padding-after);
		padding-inline-start: var(--inset-size);
		border-radius: var(--wrapper-border-radius) var(--wrapper-border-radius) 0 0;
	}

	atlas-input-chip-group [slot="chips-label"] {
		font-size: var(--chips-label-font-size);
	}

	atlas-input-chip-group::part(chips-label),
	atlas-input-chip-group::part(chips-scroll-view) {
		background-color: var(--atlas-input-common-background-color);
	}

	atlas-input-chip-group::part(chips-scroll-view) {
		grid-area: chips;
		display: block;

		padding-inline: var(--inset-size);
		padding-block-end: var(--inset-size);
		padding-block-end: var(--inset-size);

		border-inline: var(--wrapper-border);
		border-block-end: var(--wrapper-border);

		border-radius: 0 0 var(--wrapper-border-radius) var(--wrapper-border-radius);

		min-height: var(--chips-min-height);
		max-height: var(--chips-max-height);
	}

	atlas-input-chip-group::part(chips) {
		display: block;
		height: 100%;
		overflow-y: auto;
	}

	atlas-input-chip-group > atlas-chip-group:not([slot]) {
		padding-block: 0;
	}

	atlas-input-chip-group::part(controls) {
		grid-area: controls;
		display: block;
	}

	atlas-input-chip-group::part(input) {
		grid-area: input;
		display: block;
	}

	atlas-input-chip-group::part(suggestion-scroll-view) {
		grid-area: suggestion;
		display: none;
	}

	atlas-input-chip-group::part(suggestion) {
		display: inherit;
	}

	atlas-input-chip-group[show-suggestion]::part(suggestion-scroll-view) {
		display: block;
		grid-area: controls / chips / span 3;
		isolation: isolate;
		position: relative;
	}

	atlas-input-chip-group[show-suggestion]::part(suggestion) {
		padding-inline: var(--suggestion-inline-padding);

		max-width: 100%;
		min-height: var(--suggestion-min-height);
		max-height: var(--suggestion-max-height);
		overflow-y: auto;

		border: none;
		box-shadow: var(--suggestion-box-shadow);
		background-color: var(--atlas-color-white);
		border-radius: var(--atlas-input-common-border-radius);
	}

	atlas-input-chip-group[show-suggestion]::part(suggestion-scroll-view)::after {
		content: "";
		background-color: var(--atlas-color-white);
		opacity: var(--backdrop-opacity);
		height: var(--backdrop-height);

		position: absolute;
		bottom: 0;
		top: 0;
		left: 0;
		right: 0;
		z-index: -1;
	}
}
