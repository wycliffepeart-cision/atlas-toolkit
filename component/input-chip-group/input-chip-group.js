import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var InputChipGroupTemplate = "<template>\n  <slot name=\"label\" part=\"label\"></slot>\n  <slot name=\"chips-label\" part=\"chips-label\"></slot>\n  <div part=\"chips-scroll-view\">\n    <slot part=\"chips\"></slot>\n  </div>\n  <slot name=\"controls\" part=\"controls\"></slot>\n  <slot name=\"input\" part=\"input\"></slot>\n  <div part=\"suggestion-scroll-view\">\n    <slot name=\"suggestion\" part=\"suggestion\"></slot>\n  </div>\n</template>\n";

let InputChipGroup = class InputChipGroup extends HTMLElement {
};
InputChipGroup = __decorate([
    Template(InputChipGroupTemplate)
], InputChipGroup);
if (!customElements.get('atlas-input-chip-group'))
    customElements.define('atlas-input-chip-group', InputChipGroup);

export { InputChipGroup };
