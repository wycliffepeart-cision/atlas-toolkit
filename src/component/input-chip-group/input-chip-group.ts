import { Template } from '../../decorators/template';
import InputChipGroupTemplate from './input-chip-group.html';

@Template(InputChipGroupTemplate)
export class InputChipGroup extends HTMLElement {}

if (!customElements.get('atlas-input-chip-group')) customElements.define('atlas-input-chip-group', InputChipGroup);