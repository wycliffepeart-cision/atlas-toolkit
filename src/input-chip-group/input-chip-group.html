<template>
  <slot name="label" part="label"></slot>
  <slot name="chips-label" part="chips-label"></slot>
  <div part="chips-scroll-view">
    <slot part="chips"></slot>
  </div>
  <slot name="controls" part="controls"></slot>
  <slot name="input" part="input"></slot>
  <div part="suggestion-scroll-view">
    <slot name="suggestion" part="suggestion"></slot>
  </div>
</template>
