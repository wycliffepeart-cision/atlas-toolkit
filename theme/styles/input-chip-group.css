@layer atlas {
	atlas-input-chip-group {
		--inset-size: 1em;

		--wrapper-border-radius: var(--atlas-input-common-border-radius);
		--wrapper-border: var(--atlas-input-common-border);

		--chips-label-font-size: var(--atlas-input-common-label-font-size);
		--chips-label-color: var(--atlas-input-common-label-color);
		--chips-label-padding-before: var(--atlas-input-common-label-block-spacing);

		--chips-min-height: calc(var(--atlas-chip-body-height) * 5);
		--chips-max-height: calc(var(--atlas-chip-body-height) * 9);

		--suggestion-min-height: calc(var(--atlas-chip-body-height) * 2);
		--suggestion-max-height: calc(var(--atlas-chip-body-height) * 5);
		--suggestion-box-shadow: var(--atlas-input-common-box-shadow);
		--suggestion-block-start-margin: -1em;
		--suggestion-inline-padding: 0.5em;

		--overall-control-padding: 0.5em;
		--overall-margin-block-end: 3em;

		--backdrop-opacity: 0.7;
		--backdrop-height: 100%;

		display: grid;
		grid-template-columns: 1fr;
		grid-template-areas:
			"label"
			"input"
			"suggestion"
			"controls"
			"chips-label"
			"chips";

		width: 100%;
		padding-block: var(--overall-control-padding);
		margin-block-end: var(--overall-margin-block-end);
	}

	atlas-input-chip-group::part(label) {
		grid-area: label;
		display: block;

		font-size: var(--atlas-input-common-label-font-size);
		margin-block-end: var(--atlas-input-common-label-block-spacing);
	}

	atlas-input-chip-group::part(chips-label) {
		grid-area: chips-label;
		display: block;

		color: var(--chips-label-color);

		border-inline: var(--wrapper-border);
		border-block-start: var(--wrapper-border);

		padding-block-start: var(--chips-label-padding-before);
		padding-inline-start: var(--inset-size);

		border-radius: var(--wrapper-border-radius) var(--wrapper-border-radius) 0 0;
	}

	atlas-input-chip-group [slot="chips-label"] {
		font-size: var(--chips-label-font-size);
	}

	atlas-input-chip-group::part(chips-label),
	atlas-input-chip-group::part(chips-scroll-view) {
		background-color: var(--atlas-input-common-background-color);
	}

	atlas-input-chip-group::part(chips-scroll-view) {
		grid-area: chips;
		display: block;

		padding-inline: var(--inset-size);
		padding-block-start: var(--inset-size);
		padding-block-end: var(--inset-size);

		border-inline: var(--wrapper-border);
		border-block-end: var(--wrapper-border);

		border-radius: 0 0 var(--wrapper-border-radius) var(--wrapper-border-radius);

		min-height: var(--chips-min-height);
		max-height: var(--chips-max-height);
	}

	atlas-input-chip-group::part(chips) {
		display: block;
		height: 100%;
		overflow-y: auto;
	}

	atlas-input-chip-group::part(controls) {
		grid-area: controls;
		display: block;
	}

	atlas-input-chip-group::part(input) {
		grid-area: input;
		display: block;
	}

	atlas-input-chip-group::part(suggestion-scroll-view) {
		grid-area: suggestion;
		display: none;
	}

	atlas-input-chip-group::part(suggestion) {
		display: inherit;
	}

	atlas-input-chip-group[show-suggestion]::part(suggestion-scroll-view) {
		display: block;
		grid-area: controls / chips / span 3;
		isolation: isolate;
		position: relative;
	}

	atlas-input-chip-group[show-suggestion]::part(suggestion) {
		padding-inline: var(--suggestion-inline-padding);
		margin-block-start: var(--suggestion-block-start-margin);

		max-width: 100%;
		min-height: var(--suggestion-min-height);
		max-height: var(--suggestion-max-height);
		overflow-y: auto;

		border: none;
		box-shadow: var(--suggestion-box-shadow);
		background-color: var(--atlas-color-white);
		border-radius: var(--atlas-input-common-border-radius);
	}

	atlas-input-chip-group[show-suggestion]::part(suggestion-scroll-view)::after {
		content: "";
		background-color: var(--atlas-color-white);
		opacity: var(--backdrop-opacity);
		height: var(--backdrop-height);

		position: absolute;
		bottom: 0;
		top: 0;
		left: 0;
		right: 0;
		z-index: -1;
	}
}
