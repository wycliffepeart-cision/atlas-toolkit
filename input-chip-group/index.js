import { InputChipGroup } from './input-chip-group.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-input-chip-group'))
    customElements.define('atlas-input-chip-group', InputChipGroup);
