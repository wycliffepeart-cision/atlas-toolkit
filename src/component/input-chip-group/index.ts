import { InputChipGroup } from './input-chip-group';

if (!customElements.get('atlas-input-chip-group')) customElements.define('atlas-input-chip-group', InputChipGroup);
