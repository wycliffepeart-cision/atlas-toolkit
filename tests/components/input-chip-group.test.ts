import '../../src/component/input-chip-group/input-chip-group';

const INPUT_CHIP_GROUP_TAG = 'atlas-input-chip-group';

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Input', () => {
  describe('Chip Group', () => {
    it('exists', async () => {
      await customElements.whenDefined(INPUT_CHIP_GROUP_TAG);
      const dashboardElement = document.createElement(INPUT_CHIP_GROUP_TAG);
      document.body.appendChild(dashboardElement);

      (expect(dashboardElement) as any).toBeInTheDocument();
    });

    it('matches simple snapshot', () => {
      const elem = document.createElement(INPUT_CHIP_GROUP_TAG);
      document.body.appendChild(elem);

      (expect(elem) as any).toContainHTML(`<atlas-input-chip-group />`);
      (expect(elem) as any).toBeEmptyDOMElement();

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });

    it('matches slotted chips snapshot', () => {
      const elem = document.createElement(INPUT_CHIP_GROUP_TAG);
      document.body.appendChild(elem);

      const chips = document.createElement('li');
      chips.setAttribute('slot', 'chips');
      elem.appendChild(chips);

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });
  });
});
