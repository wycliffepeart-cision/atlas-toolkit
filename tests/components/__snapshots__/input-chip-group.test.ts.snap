// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Chip Group matches simple snapshot 1`] = `
<slot name="label"
      part="label"
>
</slot>
<slot name="chips-label"
      part="chips-label"
>
</slot>
<div part="chips-scroll-view">
  <slot part="chips">
  </slot>
</div>
<slot name="controls"
      part="controls"
>
</slot>
<slot name="input"
      part="input"
>
</slot>
<div part="suggestion-scroll-view">
  <slot name="suggestion"
        part="suggestion"
  >
  </slot>
</div>
`;

exports[`Input Chip Group matches slotted chips snapshot 1`] = `
<slot name="label"
      part="label"
>
</slot>
<slot name="chips-label"
      part="chips-label"
>
</slot>
<div part="chips-scroll-view">
  <slot part="chips">
  </slot>
</div>
<slot name="controls"
      part="controls"
>
</slot>
<slot name="input"
      part="input"
>
</slot>
<div part="suggestion-scroll-view">
  <slot name="suggestion"
        part="suggestion"
  >
  </slot>
</div>
`;
